__tests__/pet.test.js

```diff
@@ -0,0 +1,171 @@
+const Pet = require('../src/pet');
+const fin = new Pet('Fin');
+const child = new Pet('JJ');
+
+
+describe('constructor', () => {
+  it('returns an object', () => {
+    expect(new Pet('Fin')).toBeInstanceOf(Object);
+  });
+
+  it('sets the name property', () => {
+    const fin = new Pet('Fin');
+
+    expect(fin.name).toEqual('Fin');
+  });
+  it('has an initial age of 0', () => {
+    const fin = new Pet('Fin');
+    expect(fin.age).toEqual(0);
+  });
+});
+
+describe('grow up', () => {
+  it('ages pet by 1', () => {
+    const fin = new Pet('Fin');
+
+    fin.growUp();
+
+    expect(fin.age).toEqual(1);
+  });
+  it('throws error if pet is dead', () => {
+    const fin = new Pet('Fin');
+
+    fin.age = 30;
+
+    expect(() => fin.growUp()).toThrow('Fin is dead :(');
```

Comment:
If the pet had another name, this message would make no sense. In the pet.js file you have this error message hardcoded, meaning your code and the test is tightly coupled to one instance of the `Pet` object, this is not a good thing and should be fixed. 

If you still wanted to fix this and keep the personalised message with the pets name in it, you could do this by updating the function under test to this:
```
    if (!this.isAlive) {
        throw new Error(`${this.name} is dead :(`);
    }
```
Then in the test, you could update it to this:
```
  it('throws error if pet is dead', () => {
    const pet = new Pet('Bob');

    pet.age = 30;

    expect(() => pet.growUp()).toThrow(`${pet.name} is dead :(`);
  });
```

Test now allows for different names and is more robust as a result